mindsdb/integrations/handlers/notion_handler/notion_handler.py

```diff
@@ -0,0 +1,219 @@
+import os
+import time
+import pandas as pd
+from collections import defaultdict
+
+from pydantic import BaseModel, Extra
+from notion_client import Client
+
+from mindsdb.utilities import log
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+
+from .notion_table import (
+    NotionBlocksTable,
+    NotionCommentsTable,
+    NotionDatabaseTable,
+    NotionPagesTable,
+)
+
+
+class NotionHandlerArgs(BaseModel):
+    target: str = None
+    ft_api_info: dict = None
+    ft_result_stats: dict = None
+    runtime: str = None
+    notion_api_token: str = None
+
+    class Config:
+        # for all args that are not expected, raise an error
+        extra = Extra.forbid
+
+
+class NotionHandler(APIHandler):
+    name = "notion"
+
+    def __init__(self, name: str, **kwargs):
+        """constructor
+        Args:
+            name (str): the handler name
+        """
+        super().__init__(name)
+
+        args = kwargs.get("connection_data", {})
+        self.connection_args = {}
+
+        handler_config = Config().get("notion_handler", {})
+        token_name = ["notion_api_token", "api_token"]
+        for k in token_name:
+            if len(self.connection_args.keys()) > 0:
+                self.connection_args[k] = self.connection_args.get(
+                    list(self.connection_args.keys())[0]
+                )
+            if k in args:
+                self.connection_args[k] = args[k]
+            elif f"NOTION_{k.upper()}" in os.environ:
+                self.connection_args[k] = os.environ[f"NOTION_{k.upper()}"]
+            elif k in handler_config:
+                self.connection_args[k] = handler_config[k]
+
+        self.api = None
+        self.is_connected = False
+
+        notion_database_data = NotionDatabaseTable(self)
+        notion_pages_data = NotionPagesTable(self)
+        notion_comments_data = NotionCommentsTable(self)
+        notion_blocks_data = NotionBlocksTable(self)
+
+        self._register_table("database", notion_database_data)
+        self._register_table("pages", notion_pages_data)
+        self._register_table("blocks", notion_blocks_data)
+        self._register_table("comments", notion_comments_data)
+
+    def connect(self, args=None, **kwargs):
+        api_token = self.connection_args["api_token"]
+        notion = Client(auth=api_token)
+        return notion
+
+    def check_connection(self) -> StatusResponse:
+        response = StatusResponse(False)
+
+        try:
+            api = self.connect()
+
+            response.success = True
+
+        except Exception as e:
+            response.error_message = (
+                f"Error connecting to Notion api: {e}. Check notion_api_token"
+            )
+            log.logger.error(response.error_message)
+
+        if response.success is True and not self.connection_args["api_token"]:
```

Comment:
I do not understand the need for this. Since you are looking for the `api_token` key in `self.connection_args` within the `connect()` method, can it be that `response.success` is `True` and `api_token` is not in `self.connection_args`?